felix/fv/qos_controls_test.go

```diff
@@ -156,6 +164,106 @@ var _ = infrastructure.DatastoreDescribe(
 
 			})
 		})
+
+		Context("With packet rate limits", func() {
+			getRules := func(felixId int) string {
+				out, err := tc.Felixes[felixId].ExecOutput("iptables-save", "-c")
+				log.Infof("iptables-save -c output:\n%v", out)
+				Expect(err).NotTo(HaveOccurred())
+				return out
+			}
+
+			It("should limit packet rate correctly", func() {
+				format.MaxLength = 1000000
+
+				By("Starting iperf3 server on workload 0")
+				serverCmd := w[0].ExecCommand("iperf3", "-s")
+				err := serverCmd.Start()
+				Expect(err).NotTo(HaveOccurred())
+
+				By("Running iperf3 client on workload 1")
+				out, err := w[1].ExecOutput("iperf3", "-c", w[0].IP, "-O5", "-M1000", "-J")
+				Expect(err).NotTo(HaveOccurred())
+				baselineRate, err := getRateFromJsonOutput(out)
```

Comment:
Note, this is in bits per sec.

Reply:
this is noted in the `Info` log a few lines below it: https://github.com/projectcalico/calico/blob/483cc0e5bce25fd56d3dcd5c67ae1ac20a7897fc/felix/fv/qos_controls_test.go#L189

should I add a comment in addition to that?

Reply:
If you add a comment to explain the 800000 just below, I don't think we need a comment for this point as well.  What was confusing was the **combination** of not explaining the 800000 _and_ not indicating the units of `getRateFromJsonOutput`.